### PR TITLE
RATIS-2060 Avoid unexpected shutdown of RaftServerProxy during membership change

### PR DESCRIPTION
**Pull request by OneSizeFitsQuorum**
see [jira](https://issues.apache.org/jira/browse/RATIS-2060)

### CODE REVIEW

**Comment by OneSizeFitsQuorum** [comment]
@szetszwo PTAL

File: ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java [hunk 1]
```diff
@@ -332,6 +332,8 @@ private boolean askForVotes(Phase phase, int round) throws InterruptedException,
         case SINGLE_MODE_PASSED:
           return true;
         case NOT_IN_CONF:
+          server.close();
+          return false;
         case SHUTDOWN:
           server.getRaftServer().close();
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java:338]: Let change it for both case.
```java
+++ b/ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java
@@ -333,7 +333,7 @@ class LeaderElection implements Runnable {
           return true;
         case NOT_IN_CONF:
         case SHUTDOWN:
-          server.getRaftServer().close();
+          server.close();
           server.getStateMachine().event().notifyServerShutdown(server.getRoleInfoProto());
           return false;
         case TIMEOUT:
```

**Review by szetszwo — COMMENTED** [review]
@OneSizeFitsQuorum , thanks for working on this!  You are right that it should shutdown the division instead of the proxy.

File: ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderElection.java]: Fixed~

**Review by SzyWilliam — APPROVED** [review]
+1 the changes look good to me.
Thanks @OneSizeFitsQuorum for the patch!
Thanks @szetszwo for reviewing the patch!

**Pull request merged**